manta-accounting/src/wallet/signer/functions.rs

```diff
@@ -198,11 +192,7 @@ where
 {
     let (_, utxo, nullifier) =
         parameters.derive_spend(authorization_context, identifier, asset.clone(), rng);
-    if let Some(index) = nullifiers
-        .iter()
-        .position(move |n| n.is_related(&nullifier))
-    {
-        nullifiers.remove(index);
+    if nullifiers.contains_item(&nullifier) {
```

Comment:
do you mind writing out the logic of why we mutate the assets/nullifiers here once and then at the end as well, to make it easier for new people to get into this part of the code.

Reply:
so we insert the assets as non-provable straight away in the merkle tree, instead of inserting them as provable and then removing the proof.

Reply:
https://github.com/Manta-Network/manta-rs/issues/348